File: array and string/minimum_size_subarray_sum.java
class Solution {
    public int minSubArrayLen(int s, int[] nums) {
        
        int i=0, j=0, sum=0, min=nums.length+1;
        
        while(i < nums.length)
        {
            sum += nums[i++];
            
            while(sum >= s)
            {
                min = Math.min(min, i-j);
                sum -= nums[j++];
            }
        }
        
        return min==nums.length+1 ? 0 : min;
    }  
}

File: array and string/minimum_size_subarray_sum.py
class Solution:
    def minSubArrayLen(self, s: int, nums: List[int]) -> int:
        i = 0
        j = 0
        minimum = len(nums) + 1
        total = 0
        
        while i < len(nums):
            total = total + nums[i]
            i += 1
            
            while total >= s:
                minimum = min(minimum, i-j)
                total = total - nums[j]
                j += 1
                
        if minimum == len(nums) + 1:
            return 0
        else:
            return minimum